lib/email/sender.rb

```diff
@@ -273,11 +273,22 @@ def add_attachments(post)
 
       email_size = 0
       post.uploads.each do |upload|
+        optimized_1X = upload.optimized_images.first
+
         if FileHelper.is_supported_image?(upload.original_filename) &&
-            !should_attach_image?(upload)
+            !should_attach_image?(upload, optimized_1X)
           next
         end
 
+        # all the checks below could be performed on an upload
+        # or an optimized image, so we extract the upload only
+        # stuff first for convinience
+        original_filename = upload.original_filename
+        upload_id = upload.id
+        if optimized_1X.present?
+          upload = optimized_1X
```

Comment:
this confused me for a moment what `upload` is in the end. Maybe to somehow distinguish them we should have something like
```ruby
post.uploads.each do |original_upload|
  ...
  attached_upload = optimized_1X || original_upload
  ...
```